js-weather/script.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+//option要素の取得（配列）
+let options = document.querySelectorAll("#select option");
+
+// ページ読み込み時
+window.onload = function () {
+  //先頭のオプションタグの値を取得
+  let firstCityName = options[0].value;
+  //使用するappID
+  let appId = "4b5774e9f3d2a07b84f0f2f88e486224";
+  //urlの取得
+  const requestUrl = "https://api.openweathermap.org/data/2.5/weather?APPID=" + appId + "&lang=ja&units=metric&q=" + firstCityName + ",jp;";
+  //Ajax通信用のオブジェクトを作成
+  let xhr = new XMLHttpRequest();
+  //通信方式とURLを設定
+  xhr.open("GET", requestUrl);
+  //通信を実行する
+  xhr.send();
+  //通信ステータスが変わったら実行される関数
+  xhr.onreadystatechange = function () {
+    //通信が完了
+    if (xhr.readyState == 4) {
+      ShowTodaysWeather(xhr.responseText);
+    }
+  }
+}
+
+//select要素のchangeイベント時
+select.addEventListener('change', function () {
+  //選択されたoption番号を取得
+  let index = this.selectedIndex;
```

Comment:
```suggestion
select.onchange = () => {
  //選択されたoption番号を取得
  const index = select.selectedIndex;
```
functionとthisは使うべきではないので、こう直しましょう

参考
[アロー関数のみにしていく](https://qiita.com/shibukawa/items/19ab5c381bbb2e09d0d9#%E3%82%A2%E3%83%AD%E3%83%BC%E9%96%A2%E6%95%B0%E3%81%AE%E3%81%BF%E3%81%AB%E3%81%97%E3%81%A6%E3%81%84%E3%81%8F)